domain/application/service/storage_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+
+	"github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	"go.uber.org/mock/gomock"
+	gc "gopkg.in/check.v1"
+
+	corestorage "github.com/juju/juju/core/storage"
+	storagetesting "github.com/juju/juju/core/storage/testing"
+	"github.com/juju/juju/core/unit"
+	unittesting "github.com/juju/juju/core/unit/testing"
+	"github.com/juju/juju/domain/application/errors"
+	loggertesting "github.com/juju/juju/internal/logger/testing"
+	"github.com/juju/juju/internal/storage"
+)
+
+type storageSuite struct {
+	testing.IsolationSuite
+
+	mockState *MockState
+
+	service *Service
+}
+
+var _ = gc.Suite(&storageSuite{})
+
+func (s *storageSuite) setupMocks(c *gc.C) *gomock.Controller {
+	ctrl := gomock.NewController(c)
+	s.mockState = NewMockState(ctrl)
+
+	s.service = &Service{
```

Comment:
Should we not have a New func for this ?

Reply:
sure